cmd/kube-proxy/app/server.go

```diff
@@ -180,6 +188,13 @@ func NewOptions() *Options {
 		scheme:      scheme.Scheme,
 		codecs:      scheme.Codecs,
 		CleanupIPVS: true,
+		OnConfigChange: func(ent fsnotify.Event) {
+			switch ent.Op {
+			// exit when ConfigFile is updated
+			case fsnotify.Write:
```

Comment:
Does this capture someone writing a temp file and renaming it, too?  That's how configmap generally works.

Reply:
> Does this capture someone writing a temp file and renaming it, too?

No. We could add it. But supporting this will cause the flag `--config` of kube-proxy outdated when the filename get renamed.